sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -530,26 +530,29 @@ static inline kernel_pid_t _next_hop_l2addr(uint8_t *l2addr, uint8_t *l2addr_len
                                             kernel_pid_t iface, ipv6_addr_t *dst,
                                             gnrc_pktsnip_t *pkt)
 {
+    kernel_pid_t found_iface;
```

Comment:
If you initialize this with `KERNEL_PID_UNDEF`, you can save the `#else` part in line 540/541.